content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1107 @@
+
+```python
+# make sure train_images_dir is pointing to the .tif images
+!python data_utils/convert_geojson_to_coco.py --train_images_dir xview_dataset/train_images/ \
+  --train_images_dir_rgb xview_dataset/train_images_rgb/ \
+  --train_geojson_path xview_dataset/xView_train.geojson \
+  --output_dir xview_dataset/ \
+  --train_split_rate 0.75 \
+  --category_id_remapping data_utils/category_id_mapping.json \
+  --xview_class_labels data_utils/xview_class_labels.txt
+```
+
+## 3. Slicing/Tiling the Dataset
+
+Here we are using the SAHI library to slice our large satellite images. Satellite images can be up to 50k^2 pixels in size, which wouldnt fit in GPU memory. We alleviate this problem by slicing the image. 
+
+```python
+!python data_utils/slice_coco.py --image_dir xview_dataset/train_images_rgb/ \
+  --train_dataset_json_path xview_dataset/train.json \
+  --val_dataset_json_path xview_dataset/val.json \
+  --slice_size 300 \
+  --overlap_ratio 0.2 \
+  --ignore_negative_samples True \
+  --min_area_ratio 0.1 \
+  --output_train_dir xview_dataset/train_images_rgb_no_neg/ \
+  --output_val_dir xview_dataset/val_images_rgb_no_neg/
+```
+
+## 4. Upload to s3 bucket to support distributed training
+
+We will now upload our exported data to a publically accessible S3 bucket. This will enable for a large scale distributed experiment to have access to the dataset without installing the dataset on device. 
+View these links to learn how to upload your dataset to an S3 bucket. Review the `S3Backend` class in `data.py`
+
+* https://docs.determined.ai/latest/training/load-model-data.html#streaming-from-object-storage
+* https://codingsight.com/upload-files-to-aws-s3-with-the-aws-cli/
+
+Once you create an S3 bucket that is publically accessible, here are example commands to upload the preprocessed dataset to S3:
+
+* `aws s3 cp --recursive xview_dataset/train_sliced_no_neg/   s3://determined-ai-xview-coco-dataset/train_sliced_no_neg`
+* `aws s3 cp --recursive xview_dataset/val_sliced_no_neg/   s3://determined-ai-xview-coco-dataset/val_sliced_no_neg`
+
+Our satellite imagery data is in an S3 bucket and is prepped for distributed training, so now we can progress to model training and inference via the NGC Container.  
+
+# Part 3: End-to-End Example training object detection model using NVIDIA Pytorch Container from NGC
+
+## Training and Inference via NGC Container
+
+This notebook walks you each step to train a model using containers from the NGC Catalog. We chose the GPU optimized Pytorch container as an example. The basics of working with docker containers apply to all NGC containers.
+
+We will show you how to:
+
+* Execute training a object detection on satellite imagery using TensorFlow and Jupyter Notebook
+* Run inference on a trained object detection model using the SAHI library
+
+Note this Object Detection demo is based on https://github.com/pytorch/vision/tree/v0.11.3 and ngc docker image `nvcr.io/nvidia/pytorch:21.11-py3`
```

Comment:
Duplicate: Same note as in Installation section, part 1. Do you ened to repeat the information here?